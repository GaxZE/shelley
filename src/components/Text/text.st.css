/** Text styles */
.root {
  -st-states: truncate;
}

/* Variants */
.v0 {}

.v1 {}

.v2 {}

.v3 {}

.v4 {}

.v5 {}

.v6 {}

.v7 {}

.v8 {}

.v9 {}

.v10 {}

.v11 {}

.v12 {}

.v13 {}

.v14 {}

/* Extras */
.blockquote {}

.uppercase {}

.underline {}

.bold {}

/* State selectors */
.root:truncate {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}
