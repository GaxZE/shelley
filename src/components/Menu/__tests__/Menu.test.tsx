import React from "react";
import {
  Menu,
  MenuList,
  MenuButton,
  MenuItem,
  MenuItems,
  MenuPopover
} from "../Menu";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Link from "gatsby-link";

import Text from "../../Text/Text";

const buttonRef = React.createRef<HTMLButtonElement>();
const BigButtsString = "I love big buttons and I cannot lie...";

describe("Menu", () => {
  // @todo fix jestconfig to map ONLY CSS files to identity-obj-proxy to support Reach CSS check.
  it("supports Reach Menu Button via #as", () => {
    const tree = renderer
      .create(
        <Menu>
          <MenuButton data-testid="button-data-testid" ref={buttonRef}>
            {BigButtsString}
          </MenuButton>
          <MenuList>
            <Text as="label" uppercase vol={1}>
              Change status
            </Text>
            <MenuItem disabled onSelect={() => alert("selected!")}>
              Publish
            </MenuItem>
            <MenuItem onSelect={() => alert("selected!")}>Archive</MenuItem>
            <MenuItem onSelect={() => alert("selected!")}>Delete</MenuItem>
          </MenuList>
        </Menu>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
