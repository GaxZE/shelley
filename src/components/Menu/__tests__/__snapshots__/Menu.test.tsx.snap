// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu supports Reach Menu Button via #as 1`] = `
Array [
  <button
    aria-controls="menu--1"
    aria-haspopup={true}
    className="Button2465625684__root Button2465625684---tone-1-1 Button2465625684---variant-1-1 Button2465625684---vol-1-3"
    data-reach-menu-button=""
    data-testid="button-data-testid"
    id="menu-button--menu"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    type="button"
  >
    <span
      className="Button2465625684__inner"
    >
      I love big buttons and I cannot lie...
    </span>
  </button>,
  <span />,
]
`;
